jdaviz/configs/imviz/plugins/line_profile_xy/line_profile_xy.py

```diff
@@ -0,0 +1,128 @@
+    template_file = __file__, "line_profile_xy.vue"
+    viewer_items = List([]).tag(sync=True)
+    selected_viewer = Unicode("").tag(sync=True)
+    plot_available = Bool(False).tag(sync=True)
+    selected_x = Any('').tag(sync=True)
+    selected_y = Any('').tag(sync=True)
+    line_plot_across_x = Any('').tag(sync=True, **widget_serialization)
+    line_plot_across_y = Any('').tag(sync=True, **widget_serialization)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._default_viewer = f'{self.app.config}-0'
+
+        self.hub.subscribe(self, ViewerAddedMessage, handler=self._on_viewers_changed)
+        self.hub.subscribe(self, ViewerRemovedMessage, handler=self._on_viewers_changed)
+
+        self._on_viewers_changed()  # Populate it on start-up
+
+    def _on_viewers_changed(self, msg=None):
+        self.viewer_items = self.app.get_viewer_ids()
+
+        # Selected viewer was removed but Imviz always has a default viewer to fall back on.
+        if self.selected_viewer not in self.viewer_items:
+            self.selected_viewer = self._default_viewer
+
+    def reset_results(self):
+        self.plot_available = False
+        self.line_plot_across_x = ''
+        self.line_plot_across_y = ''
+        bqplt.clear()
+
+    # This is also triggered from viewer code.
+    @observe("selected_viewer")
+    def vue_draw_plot(self, *args, **kwargs):
+        """Draw line profile plots for given Data across given X and Y indices (0-indexed)."""
+        if not self.selected_x or not self.selected_y or not self.plugin_opened:
+            return
+
+        viewer = self.app.get_viewer_by_id(self.selected_viewer)
+        i = get_top_layer_index(viewer)
+        data = viewer.state.layers[i].layer
+
+        if isinstance(self.selected_x, str):
+            x = int(round(float(self.selected_x)))
+        else:
+            x = int(round(self.selected_x))
+
+        if isinstance(self.selected_y, str):
+            y = int(round(float(self.selected_y)))
+        else:
+            y = int(round(self.selected_y))
+
+        nx = data.shape[1]
+        ny = data.shape[0]
+        if x < 0 or y < 0 or x >= nx or y >= ny:
+            self.reset_results()
+            return
+
+        x_min, y_min, x_max, y_max = viewer._get_zoom_limits(data)
+        comp = data.get_component(data.main_components[0])
+        if comp.units:
+            y_label = comp.units
+        else:
+            y_label = 'Value'
+
+        _bqplt_clear_all()
+
+        fig_x = bqplt.figure(1, title=f'X={x}',
+                             fig_margin={'top': 60, 'bottom': 60, 'left': 40, 'right': 10},
+                             title_style={'font-size': '12px'})
+        bqplt.plot(comp.data[:, x], colors='gray', figure=fig_x)
+        bqplt.xlim(y_min, y_max)
+        y_min = int(y_min)
+        if y_min < 0:
+            y_min = 0
+        y_max = int(y_max)
+        if y_max > ny:
+            y_max = ny

```

Comment:
```suggestion
        y_min = max(int(y_min), 0)
        y_max = min(int(y_max), 0)
```

Reply:
Wait... Did you mean `y_max = min(int(y_max), ny)`?

Reply:
Woops, yes!

Reply:
I already applied the correct version in the last commit and gave you co-author credit. 😸